File: model/src/main/kotlin/net/dungeonhub/model/discord_user/DiscordUserModel.java
package net.dungeonhub.model.discord_user;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import me.taubsie.dungeonhub.common.DungeonHubService;
import me.taubsie.dungeonhub.common.entity.model.Model;

import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
public class DiscordUserModel implements Model {
    private final long id;
    private UUID minecraftId;

    public static DiscordUserModel fromJson(String json) {
        return DungeonHubService.getInstance()
                .getGson()
                .fromJson(json, DiscordUserModel.class);
    }
}

File: model/src/main/kotlin/net/dungeonhub/model/discord_user/DiscordUserModel.kt
package net.dungeonhub.model.discord_user

import net.dungeonhub.service.MoshiService
import net.dungeonhub.structure.model.UpdateableModel
import java.util.*

class DiscordUserModel(
    val id: Long,
    val minecraftId: UUID?
) : UpdateableModel<DiscordUserUpdateModel, DiscordUserModel> {
    companion object {
        fun fromJson(json: String): DiscordUserModel {
            return MoshiService.moshi.adapter(DiscordUserModel::class.java).fromJson(json)!!
        }
    }

    override fun getUpdateModel(): DiscordUserUpdateModel {
        return DiscordUserUpdateModel(null)
    }
}